code/modules/mob/mob_movement.dm

```diff
@@ -178,11 +178,17 @@
 		if(mob.throwing)
 			mob.throwing.finalize(FALSE)
 
-	for(var/obj/O in mob.contents|mob.held_items)
-		O.on_mob_move(direct, src)
-
 	return .
 
+/mob/Moved(oldLoc, dir)
+	. = ..()
+	for(var/obj/O in contents|held_items)
+		O.on_mob_move(dir, src, oldLoc)

```

Comment:
the carbon one only works on internal organs as internal organs aren't defined in mob